src/main/java/bio/terra/cli/app/DockerCommandRunner.java

```diff
@@ -57,16 +64,45 @@ protected String wrapCommandInSetupCleanup(List<String> command) {
    */
   protected int runToolCommandImpl(String command, Map<String, String> envVars)
       throws PassthroughException {
-    // set the path to the pet SA key file, which may be different on the container vs the host
-    envVars.put("GOOGLE_APPLICATION_CREDENTIALS", getPetSaKeyFileOnContainer().toString());
-
     // mount the global context directory and the current working directory to the container
     //  e.g. global context dir (host) $HOME/.terra -> (container) CONTAINER_HOME_DIR/.terra
     //       current working dir (host) /Users/mm/workspace123 -> (container) CONTAINER_HOME_DIR
     Map<Path, Path> bindMounts = new HashMap<>();
     bindMounts.put(getGlobalContextDirOnContainer(), Context.getContextDir());
     bindMounts.put(Path.of(CONTAINER_WORKING_DIR), Path.of(System.getProperty("user.dir")));
 
+    // mount the gcloud config directory to the container
+    // e.g. gcloud config dir (host) $HOME/.config/gcloud -> (container)
+    // CONTAINER_HOME_DIR/.config/gcloud
+    Path gcloudConfigDir = Path.of(System.getProperty("user.home"), ".config/gcloud");
+    if (gcloudConfigDir.toFile().exists() && gcloudConfigDir.toFile().isDirectory()) {
+      bindMounts.put(Path.of(CONTAINER_HOME_DIR, ".config/gcloud"), gcloudConfigDir);
+    }
+
+    // check if the testing flag is set to a key file
+    Optional<Path> adcBackingFile = getOverrideCredentialsFileForTesting();
+    if (adcBackingFile.isEmpty()) {
+      // testing flag is not set, this is normal operation
+      // application default credentials must be set to the user or their pet SA
+      AppDefaultCredentialUtils.throwIfADCDontMatchContext();
+      adcBackingFile = AppDefaultCredentialUtils.getADCBackingFile();
+    }
+
+    // mount the application default credentials file to the container
+    // e.g. ADC file (host) $HOME/pet-sa-key.json -> (container)
+    // CONTAINER_HOME_DIR/.terra/pet-keys/[user id]/application_default_credentials.json
+    if (adcBackingFile.isPresent()) {
```

Comment:
nit: this could be an `else` clause of the `if (adcBackingFile.isEmpty())` block above

Reply:
Is that true given the block above sets `adcBackingFile`?